resource-aggregate/events/deviceMetadataSnapshotTaken.go

```diff
@@ -93,8 +93,22 @@ func (e *DeviceMetadataSnapshotTaken) HandleDeviceMetadataSnapshotTaken(ctx cont
 	return nil
 }
 
-func (e *DeviceMetadataSnapshotTaken) HandleDeviceMetadataUpdatePending(ctx context.Context, updatePending *DeviceMetadataUpdatePending) error {
+func (e *DeviceMetadataSnapshotTaken) HandleDeviceMetadataUpdatePending(ctx context.Context, updatePending *DeviceMetadataUpdatePending, isCmd bool) error {
+	now := time.Now()
+	if !updatePending.IsValid(now) {
+		if isCmd {
+			// for command we want to report that we are not able process request.
+			return status.Errorf(codes.InvalidArgument, "valid_until(`%v`) is less than now(`%v`)", updatePending.ValidUntilTime(), now)
+		}
+		e.DeviceId = updatePending.GetDeviceId()
+		e.EventMetadata = updatePending.GetEventMetadata()
+		// for events from eventstore we do nothing
+		return nil
+	}
 	for _, event := range e.GetUpdatePendings() {
+		if !event.IsValid(now) {
+			continue
```

Comment:
Don't forget to document that expired pending commands are removed only when the new snapshot is created.

Reply:
https://github.com/plgd-dev/cloud/blob/validUntilCommands/resource-aggregate/pb/events.proto#L114